max/kernels/src/comm/allreduce.mojo

```diff
@@ -153,8 +151,8 @@ comptime _target_address_space = AddressSpace.GLOBAL if is_amd_gpu() else Addres
 fn _naive_reduce_kernel[
     dtype: DType
 ](
-    dst_buf: UnsafePointer[Scalar[dtype]],
-    src_buf: UnsafePointer[Scalar[dtype]],
+    dst_buf: UnsafePointer[mut=True, Scalar[dtype], MutAnyOrigin],
+    src_buf: UnsafePointer[mut=False, Scalar[dtype], MutAnyOrigin],
```

Comment:
When `UnsafePointer` is used as a function argument - you can leave the `origin` unbound, instead of forcing it to be the "any" origin.
```mojo
    dst_buf: UnsafePointer[mut=True, Scalar[dtype]],
    src_buf: UnsafePointer[mut=False, Scalar[dtype]],
```